esmvalcore/_diagnostics.py

```diff
@@ -0,0 +1,110 @@
+"""Diagnostics and tags management."""
+import logging
+import os
+from pathlib import Path
+
+import yaml
+
+logger = logging.getLogger(__name__)
+
+
+class Diagnostics:
+    """Manages the location of the recipes and diagnostics.
+
+    The directory structure is defined below:
+
+    .. code-block
+
+        <path>
+        ├── config-references.yml [-> tags_config]
+        ├── diag_scripts
+        ├── recipes
+        └── references
+    """
+
+    def __init__(self, path):
+        self.path = Path(path)
+
+    @property
+    def recipes(self):
+        """Return the location of the recipes."""
+        return self.path / 'recipes'
+
+    @property
+    def references(self):
+        """Return location of the references (bibtex files)."""
+        return self.path / 'references'
+
+    @property
+    def tags_config(self):
+        """Return location of the tags config."""
+        return self.path / 'config-references.yml'
+
+    @property
+    def scripts(self):
+        """Return location of diagnostic scripts."""
+        return self.path / 'diag_scripts'
+
+    def load_tags(self):
+        """Load the tags config into an instance of ``TagsManager``."""
+        return TagsManager.from_file(self.tags_config)
+
+    @classmethod
+    def find(cls):
+        """Try to find installed diagnostic scripts."""
+        try:
+            import esmvaltool
+        except ImportError:
+            path = Path.cwd()
+        else:
+            # avoid a crash when there is a directory called
+            # 'esmvaltool' that is not a Python package
+            if esmvaltool.__file__ is None:
+                path = Path.cwd()
+            else:
+                path = Path(esmvaltool.__file__).absolute().parent
+        return cls(path)
```

Comment:
it'd be good to add a couple logging statements here a la `"Using diagnostics from esmvaltool installation in %s path", path` at least as DEBUG messages if not INFO

Reply:
I can do that, but this message will not show up in many cases, because `DIAGNOSTICS` gets imported before the logger is initialized.